sdk/cloudmachine/Azure.CloudMachine/src/AIFoundary/AIFoundryClient.cs

```diff
@@ -0,0 +1,255 @@
+                    }
+                }
+                finally
+                {
+                    _connectionCacheLock.ExitWriteLock();
+                }
+            }
+
+            // If the connection uses API key auth, validate and add if needed.
+            if (connection.Properties is ConnectionPropertiesApiKeyAuth apiKeyAuthProperties)
+            {
+                if (string.IsNullOrWhiteSpace(apiKeyAuthProperties.Target))
+                {
+                    throw new ArgumentException(
+                        $"The API key authentication target URI is missing or invalid for {connectionId}.");
+                }
+
+                if (apiKeyAuthProperties.Credentials == null
+                    || string.IsNullOrWhiteSpace(apiKeyAuthProperties.Credentials.Key))
+                {
+                    throw new ArgumentException($"The API key is missing or invalid for {connectionId}.");
+                }
+
+                // Build the new connection object.
+                var newConnection = new ClientConnection(connectionId, apiKeyAuthProperties.Target, apiKeyAuthProperties.Credentials.Key);
+
+                // Now we need to re-check and possibly add to Connections under a write lock.
+                _connectionsLock.EnterUpgradeableReadLock();
+                try
+                {
+                    if (Connections.Contains(connectionId))
+                    {
+                        return Connections[connectionId];
+                    }
+                    else
+                    {
+                        _connectionsLock.EnterWriteLock();
+                        try
+                        {
+                            // Double-check again after acquiring write lock.
+                            if (!Connections.Contains(connectionId))
+                            {
+                                Connections.Add(newConnection);
+                                return newConnection;
+                            }
+                            else
+                            {
+                                return Connections[connectionId]; // Some thread beat us to it.
+                            }
+                        }
+                        finally
+                        {
+                            _connectionsLock.ExitWriteLock();
+                        }
+                    }
+                }
+                finally
+                {
+                    _connectionsLock.ExitUpgradeableReadLock();
+                }
+            }
+            else
+            {
+                throw new ArgumentException(
+                    $"Cannot connect with {connectionId}! Ensure valid API key authentication."
+                );
+            }
+        }
+
+        private ConnectionType GetConnectionTypeFromId(string connectionId)
+        {
+            switch (connectionId)
+            {
+                case var id when
```

Comment:
why do you need the "id' here? Can you just use a simple switch-case? I think you will need to use string literals for the cases, but I think it's OK. Alternatively, just create a dictionary from string to ConnectionType and do lookup. Such lookup is approximately what the compiler generates for a simple swith-case

Reply:
I updated the method to use a simple switch on string literals because it’s more straightforward and readable. Each case clearly shows which fully qualified type name maps to which `ConnectionType`, without the overhead of creating or maintaining a separate dictionary.